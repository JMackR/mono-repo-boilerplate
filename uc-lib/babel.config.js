module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
