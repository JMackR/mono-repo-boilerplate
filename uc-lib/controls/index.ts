export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
