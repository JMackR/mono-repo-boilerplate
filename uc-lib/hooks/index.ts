export * from "./use-infinite-scroll"
export * from "./selectable"
export * from "./screen-provider"
export * from "./loading"
