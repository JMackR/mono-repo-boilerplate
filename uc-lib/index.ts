export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
