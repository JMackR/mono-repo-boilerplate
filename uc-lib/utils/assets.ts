import { PixelRatio } from "react-native"

const pixelDensityExtension = () => {
  const pixelRatio = PixelRatio.get()
  if (pixelRatio >= 3) {
    return "@3x"
  } else if (pixelRatio >= 2) {
    return "@2x"
  } else {
    return ""
  }
}

/**
 * Constructs and returns a formatted cdn url
 * @param path to asset including extension (e.g. "mojis/assassin-64.png")
 * @return the constructed cdn url
 */
export const cdnUrl = (path: string) => {
  const prefix = "https://wrkshp-cdn.com/com.nextmusic/"
  const url = `${prefix}${path}`
  const index = url.lastIndexOf(".")
  return url.slice(0, index) + pixelDensityExtension() + url.slice(index)
}
