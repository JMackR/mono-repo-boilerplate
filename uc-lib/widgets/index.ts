export * from "./account"
export * from "./attribute-list"
export * from "./breadcrumb"
export * from "./dialog/"
export * from "./empty-state"
export * from "./expanding-text-section"
export * from "./flexible-row"
export * from "./full-screen-spinner"
export * from "./inbox-row"
export * from "./indicator-dots"
export * from "./item-preview"
export * from "./listing-photo-list"
export * from "./pill-list"
export * from "./segmented-control"
export * from "./selectable-row"
export * from "./tooltip"
export * from "./user-profile"
export * from "./search"
