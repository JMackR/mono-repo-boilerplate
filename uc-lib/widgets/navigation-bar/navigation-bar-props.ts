import { LocalSVGSource } from "uc-lib/controls"
import { TextColors } from "../../themes"

export interface NavigationBarItem {
  title?: string
  image?: object
  colorTint?: string
  icon?: LocalSVGSource
  pressHandler: () => void
  /**
   * A color tint that takes precedence over the navigation bar's `barItemsTint` prop.
   * @default 'brand'
   */
  tint?: keyof TextColors
  /**
   * Used to locate this view in end-to-end tests.
   */
  testID?: string
}

/**
 * TODO bring back in the required colorTint prop
 */
export interface NavigationBarProps {
  title?: string
  image?: object
  colorTint?: string
  backgroundColor?: string
  leftItems?: NavigationBarItem[]
  rightItems?: NavigationBarItem[]
  isRootNavBar?: boolean
  /**
   * A color tint that applies to all navigation bar items.
   * @default 'brand'
   */
  barItemsTint?: keyof TextColors
  /**
   * Used to locate this view in end-to-end tests.
   */
  testID?: string
}
