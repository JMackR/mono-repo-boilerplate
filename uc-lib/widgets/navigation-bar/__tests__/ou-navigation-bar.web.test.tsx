import React from "react"
import * as renderer from "react-test-renderer"
import { NavigationBar } from "uc-lib"
import { NavigationBarProps, NavigationBarItem } from "../navigation-bar-props"
import { LocationPinFill } from "uc-lib/assets"

describe("<NavigationBar />", () => {
  const createComponent = (props: NavigationBarProps) => <NavigationBar {...props} />
  const createItem = (title: string): NavigationBarItem => {
    return { title: title, pressHandler: () => {} }
  }
  const iconItem = { icon: LocationPinFill, pressHandler: () => {} }

  describe("Snapshot", () => {
    it("should render only a title", () => {
      const tree = renderer.create(createComponent({ title: "Page title" }))
      expect(tree).toMatchSnapshot()
    })
    it("should render one left item", () => {
      const tree = renderer.create(
        createComponent({
          title: "Page title",
          leftItems: [createItem("Left1")],
        }),
      )
      expect(tree).toMatchSnapshot()
    })
    it("should render multiple left items", () => {
      const tree = renderer.create(
        createComponent({
          title: "Page title",
          leftItems: [createItem("Left1"), createItem("Left2")],
        }),
      )
      expect(tree).toMatchSnapshot()
    })
    it("should render multiple left items with icon", () => {
      const tree = renderer.create(
        createComponent({
          title: "Page title",
          leftItems: [createItem("Left1"), iconItem],
        }),
      )
      expect(tree).toMatchSnapshot()
    })
    it("should render one right item", () => {
      const tree = renderer.create(
        createComponent({
          title: "Page title",
          rightItems: [createItem("Right1")],
        }),
      )
      expect(tree).toMatchSnapshot()
    })
    it("should render multiple right items", () => {
      const tree = renderer.create(
        createComponent({
          title: "Page title",
          rightItems: [createItem("Right1"), createItem("Right2")],
        }),
      )
      expect(tree).toMatchSnapshot()
    })
    it("should render multiple right items with icons", () => {
      const tree = renderer.create(
        createComponent({
          title: "Page title",
          rightItems: [createItem("Right1"), iconItem],
        }),
      )
      expect(tree).toMatchSnapshot()
    })
  })
})
