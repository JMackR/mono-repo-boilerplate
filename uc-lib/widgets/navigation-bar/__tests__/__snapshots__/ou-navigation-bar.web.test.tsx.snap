// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> Snapshot should render multiple left items 1`] = `
.ouflex {
  display: flex;
  flex-basis: 56px;
  flex-grow: 0;
  flex-shrink: 0;
  flex-direction: row;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: 100%;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.ou {
  display: flex;
  position: absolute;
  top: 24px;
  bottom: 8px;
  left: 48px;
  right: 48px;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: center;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

.text-textStyle {
  color: #121212;
  font-size: 20px;
  font-weight: 900;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: center;
}

.ou {
  display: flex;
  position: absolute;
  top: 16px;
  bottom: 8px;
  left: 16px;
  right: undefined;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: flex-end;
  justify-content: flex-start;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

.oustack {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  width: undefined;
  max-width: undefined;
  overflow: undefined;
}

.oredibs-uclickableopacity {
  width: 100%;
  display: inline-block;
  cursor: pointer;
}

.text-textStyle {
  color: #00A87E;
  font-size: 16px;
  font-weight: bold;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

.text-textStyle:hover {
  cursor: pointer;
}

.root {
  height: undefined;
  width: 8px;
  display: inline-block;
}

<div
  className="ouflex"
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 0,
      }
    }
    type="primary"
  />
  <div
    className="ou"
  >
    <div
      className="text-textStyle"
    >
      Page title
    </div>
  </div>
  <div
    className="ou"
  >
    <div
      className="oustack"
    >
      <div
        className="oredibs-uclickableopacity"
        data-test-id="redibs-ucl.navigation-bar-item.0"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 40,
            "minWidth": 40,
          }
        }
      >
        <div
          className="text-textStyle"
        >
          Left1
        </div>
      </div>
      <div
        className="root"
      />
      <div
        className="oredibs-uclickableopacity"
        data-test-id="redibs-ucl.navigation-bar-item.1"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 40,
            "minWidth": 40,
          }
        }
      >
        <div
          className="text-textStyle"
        >
          Left2
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> Snapshot should render multiple left items with icon 1`] = `
.ouflex {
  display: flex;
  flex-basis: 56px;
  flex-grow: 0;
  flex-shrink: 0;
  flex-direction: row;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: 100%;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.ou {
  display: flex;
  position: absolute;
  top: 24px;
  bottom: 8px;
  left: 48px;
  right: 48px;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: center;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

.text-textStyle {
  color: #121212;
  font-size: 20px;
  font-weight: 900;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: center;
}

.ou {
  display: flex;
  position: absolute;
  top: 16px;
  bottom: 8px;
  left: 16px;
  right: undefined;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: flex-end;
  justify-content: flex-start;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

.oustack {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  width: undefined;
  max-width: undefined;
  overflow: undefined;
}

.oredibs-uclickableopacity {
  width: 100%;
  display: inline-block;
  cursor: pointer;
}

.text-textStyle {
  color: #00A87E;
  font-size: 16px;
  font-weight: bold;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

.text-textStyle:hover {
  cursor: pointer;
}

.root {
  height: undefined;
  width: 8px;
  display: inline-block;
}

<div
  className="ouflex"
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 0,
      }
    }
    type="primary"
  />
  <div
    className="ou"
  >
    <div
      className="text-textStyle"
    >
      Page title
    </div>
  </div>
  <div
    className="ou"
  >
    <div
      className="oustack"
    >
      <div
        className="oredibs-uclickableopacity"
        data-test-id="redibs-ucl.navigation-bar-item.0"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 40,
            "minWidth": 40,
          }
        }
      >
        <div
          className="text-textStyle"
        >
          Left1
        </div>
      </div>
      <div
        className="root"
      />
      <div
        className="oredibs-uclickableopacity"
        data-test-id="redibs-ucl.navigation-bar-item.1"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 40,
            "minWidth": 40,
          }
        }
      >
        <location-fill-pin.svg
          alwaysdark="#121212"
          alwayslight="#FFFFFF"
          basalt="#8A8A8A"
          crystal="#FFFFFF"
          crystalhover="#F7F7F7"
          crystalpressed="#EEEEEE"
          disabled="#FAFAFA"
          glacialblue="#3D98D2"
          granite="#ABABAB"
          height={24}
          larchyellow="#F4BE19"
          larchyellowhover="#F5D624"
          larchyellowpressed="#F3A70C"
          limestone="#E2E2E2"
          limestonehover="#EEEEEE"
          limestonepressed="#D0D0D0"
          obsidian="#121212"
          overlay="rgba(18,18,18,0.3)"
          paintbrushred="#E05666"
          palousegreen="#00A87E"
          palousegreenhighlight="#DEF3EC"
          palousegreenhover="#43B893"
          palousegreenpressed="#00986A"
          quartz="#FAFAFA"
          tintcolor="#00A87E"
          width={24}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> Snapshot should render multiple right items 1`] = `
.ouflex {
  display: flex;
  flex-basis: 56px;
  flex-grow: 0;
  flex-shrink: 0;
  flex-direction: row;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: 100%;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.ou {
  display: flex;
  position: absolute;
  top: 24px;
  bottom: 8px;
  left: 48px;
  right: 48px;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: center;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

.text-textStyle {
  color: #121212;
  font-size: 20px;
  font-weight: 900;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: center;
}

.oustack {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  width: undefined;
  max-width: undefined;
  overflow: undefined;
}

.oredibs-uclickableopacity {
  width: 100%;
  display: inline-block;
  cursor: pointer;
}

.text-textStyle {
  color: #00A87E;
  font-size: 16px;
  font-weight: bold;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

.text-textStyle:hover {
  cursor: pointer;
}

.root {
  height: undefined;
  width: 8px;
  display: inline-block;
}

.ou {
  display: flex;
  position: absolute;
  top: 16px;
  bottom: 8px;
  left: undefined;
  right: 16px;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row-reverse;
  align-items: flex-end;
  justify-content: flex-start;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

<div
  className="ouflex"
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 0,
      }
    }
    type="primary"
  />
  <div
    className="ou"
  >
    <div
      className="text-textStyle"
    >
      Page title
    </div>
  </div>
  <div
    className="ou"
  >
    <div
      className="oustack"
    >
      <div
        className="oredibs-uclickableopacity"
        data-test-id="redibs-ucl.navigation-bar-item.0"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row-reverse",
            "height": 40,
            "minWidth": 40,
          }
        }
      >
        <div
          className="text-textStyle"
        >
          Right1
        </div>
      </div>
      <div
        className="root"
      />
      <div
        className="oredibs-uclickableopacity"
        data-test-id="redibs-ucl.navigation-bar-item.1"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row-reverse",
            "height": 40,
            "minWidth": 40,
          }
        }
      >
        <div
          className="text-textStyle"
        >
          Right2
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> Snapshot should render multiple right items with icons 1`] = `
.ouflex {
  display: flex;
  flex-basis: 56px;
  flex-grow: 0;
  flex-shrink: 0;
  flex-direction: row;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: 100%;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.ou {
  display: flex;
  position: absolute;
  top: 24px;
  bottom: 8px;
  left: 48px;
  right: 48px;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: center;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

.text-textStyle {
  color: #121212;
  font-size: 20px;
  font-weight: 900;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: center;
}

.oustack {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  width: undefined;
  max-width: undefined;
  overflow: undefined;
}

.oredibs-uclickableopacity {
  width: 100%;
  display: inline-block;
  cursor: pointer;
}

.text-textStyle {
  color: #00A87E;
  font-size: 16px;
  font-weight: bold;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

.text-textStyle:hover {
  cursor: pointer;
}

.root {
  height: undefined;
  width: 8px;
  display: inline-block;
}

.ou {
  display: flex;
  position: absolute;
  top: 16px;
  bottom: 8px;
  left: undefined;
  right: 16px;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row-reverse;
  align-items: flex-end;
  justify-content: flex-start;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

<div
  className="ouflex"
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 0,
      }
    }
    type="primary"
  />
  <div
    className="ou"
  >
    <div
      className="text-textStyle"
    >
      Page title
    </div>
  </div>
  <div
    className="ou"
  >
    <div
      className="oustack"
    >
      <div
        className="oredibs-uclickableopacity"
        data-test-id="redibs-ucl.navigation-bar-item.0"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row-reverse",
            "height": 40,
            "minWidth": 40,
          }
        }
      >
        <div
          className="text-textStyle"
        >
          Right1
        </div>
      </div>
      <div
        className="root"
      />
      <div
        className="oredibs-uclickableopacity"
        data-test-id="redibs-ucl.navigation-bar-item.1"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row-reverse",
            "height": 40,
            "minWidth": 40,
          }
        }
      >
        <location-fill-pin.svg
          alwaysdark="#121212"
          alwayslight="#FFFFFF"
          basalt="#8A8A8A"
          crystal="#FFFFFF"
          crystalhover="#F7F7F7"
          crystalpressed="#EEEEEE"
          disabled="#FAFAFA"
          glacialblue="#3D98D2"
          granite="#ABABAB"
          height={24}
          larchyellow="#F4BE19"
          larchyellowhover="#F5D624"
          larchyellowpressed="#F3A70C"
          limestone="#E2E2E2"
          limestonehover="#EEEEEE"
          limestonepressed="#D0D0D0"
          obsidian="#121212"
          overlay="rgba(18,18,18,0.3)"
          paintbrushred="#E05666"
          palousegreen="#00A87E"
          palousegreenhighlight="#DEF3EC"
          palousegreenhover="#43B893"
          palousegreenpressed="#00986A"
          quartz="#FAFAFA"
          tintcolor="#00A87E"
          width={24}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> Snapshot should render one left item 1`] = `
.ouflex {
  display: flex;
  flex-basis: 56px;
  flex-grow: 0;
  flex-shrink: 0;
  flex-direction: row;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: 100%;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.ou {
  display: flex;
  position: absolute;
  top: 24px;
  bottom: 8px;
  left: 48px;
  right: 48px;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: center;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

.text-textStyle {
  color: #121212;
  font-size: 20px;
  font-weight: 900;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: center;
}

.ou {
  display: flex;
  position: absolute;
  top: 16px;
  bottom: 8px;
  left: 16px;
  right: undefined;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: flex-end;
  justify-content: flex-start;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

.oustack {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  width: undefined;
  max-width: undefined;
  overflow: undefined;
}

.oredibs-uclickableopacity {
  width: 100%;
  display: inline-block;
  cursor: pointer;
}

.text-textStyle {
  color: #00A87E;
  font-size: 16px;
  font-weight: bold;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

.text-textStyle:hover {
  cursor: pointer;
}

<div
  className="ouflex"
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 0,
      }
    }
    type="primary"
  />
  <div
    className="ou"
  >
    <div
      className="text-textStyle"
    >
      Page title
    </div>
  </div>
  <div
    className="ou"
  >
    <div
      className="oustack"
    >
      <div
        className="oredibs-uclickableopacity"
        data-test-id="redibs-ucl.navigation-bar-item.0"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 40,
            "minWidth": 40,
          }
        }
      >
        <div
          className="text-textStyle"
        >
          Left1
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> Snapshot should render one right item 1`] = `
.ouflex {
  display: flex;
  flex-basis: 56px;
  flex-grow: 0;
  flex-shrink: 0;
  flex-direction: row;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: 100%;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.ou {
  display: flex;
  position: absolute;
  top: 24px;
  bottom: 8px;
  left: 48px;
  right: 48px;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: center;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

.text-textStyle {
  color: #121212;
  font-size: 20px;
  font-weight: 900;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: center;
}

.oustack {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  width: undefined;
  max-width: undefined;
  overflow: undefined;
}

.oredibs-uclickableopacity {
  width: 100%;
  display: inline-block;
  cursor: pointer;
}

.text-textStyle {
  color: #00A87E;
  font-size: 16px;
  font-weight: bold;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

.text-textStyle:hover {
  cursor: pointer;
}

.ou {
  display: flex;
  position: absolute;
  top: 16px;
  bottom: 8px;
  left: undefined;
  right: 16px;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row-reverse;
  align-items: flex-end;
  justify-content: flex-start;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

<div
  className="ouflex"
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 0,
      }
    }
    type="primary"
  />
  <div
    className="ou"
  >
    <div
      className="text-textStyle"
    >
      Page title
    </div>
  </div>
  <div
    className="ou"
  >
    <div
      className="oustack"
    >
      <div
        className="oredibs-uclickableopacity"
        data-test-id="redibs-ucl.navigation-bar-item.0"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row-reverse",
            "height": 40,
            "minWidth": 40,
          }
        }
      >
        <div
          className="text-textStyle"
        >
          Right1
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<NavigationBar /> Snapshot should render only a title 1`] = `
.ouflex {
  display: flex;
  flex-basis: 56px;
  flex-grow: 0;
  flex-shrink: 0;
  flex-direction: row;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: 100%;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.ou {
  display: flex;
  position: absolute;
  top: 24px;
  bottom: 8px;
  left: 48px;
  right: 48px;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: center;
  background-color: undefined;
  z-index: 1;
  height: undefined;
  width: undefined;
}

.text-textStyle {
  color: #121212;
  font-size: 20px;
  font-weight: 900;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: center;
}

<div
  className="ouflex"
>
  <div
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 0,
      }
    }
    type="primary"
  />
  <div
    className="ou"
  >
    <div
      className="text-textStyle"
    >
      Page title
    </div>
  </div>
</div>
`;
